app/uk/gov/hmrc/servicedependencies/persistence/derived/DerivedDependencyRepository.scala

```diff
@@ -0,0 +1,133 @@
+  ) ++ DependencyScope.values.map(s => IndexModel(Indexes.ascending(s.asString + "Flag"), IndexOptions().name(s.asString + "FlagIdx")))
+) with Logging {
+
+  // automatically refreshed when given new meta data artefacts from update scheduler
+  override lazy val requiresTtlIndex = false
+
+  def addAndReplace(dependencies: Seq[MetaArtefactDependency]): Future[Unit] = {
+    collection
+        .bulkWrite(
+          dependencies.map(d =>
+            ReplaceOneModel(
+              filter = Filters.and(
+                equal("slugName",     d.slugName),
+                equal("group",        d.group),
+                equal("artefact",     d.artefact)
+              ),
+              replacement = d,
+              replaceOptions = ReplaceOptions().upsert(true)
+            )
+          )
+        ).toFuture()
+        .map(_ => ())
+  }
+
+  def find(
+            slugName: Option[String]              = None,
+            repoType: Option[RepoType]            = None,
+            group: Option[String]                 = None,
+            artefact: Option[String]              = None,
+            scopes: Option[List[DependencyScope]] = None
+          ): Future[Seq[MetaArtefactDependency]] = {
+
+    val nameFilter: Option[Bson]      = slugName.map(slugName => equal("slugName", slugName))
+    val groupFilter: Option[Bson]     = group.map(group => equal("group", group))
+    val artifactFilter: Option[Bson]  = artefact.map(artifact => equal("artefact", artifact))
+    val scopeFilter: Bson             = scopes.fold[Bson](BsonDocument())(ss => or(ss.map(scope => equal(s"${scope.asString}Flag", value = true)): _*))
+    val repoFilter: Option[Bson]      = repoType.map(repoType => equal("repoType", repoType.toString))
+
+    val filters = Seq(nameFilter, groupFilter, artifactFilter, repoFilter, Some(scopeFilter)).flatten
+
+    collection
+      .find(if (filters.isEmpty) BsonDocument() else Filters.and(filters: _*))
+      .toFuture()
+  }
+
+  def findByOtherRepository(
+                             group: Option[String]                 = None,
+                             artefact: Option[String]              = None,
+                             scopes: Option[List[DependencyScope]] = None
+                           ): Future[Seq[MetaArtefactDependency]] = {
+
+    val groupFilter: Option[Bson]     = group.map(group => equal("group", group))
+    val artifactFilter: Option[Bson]  = artefact.map(artifact => equal("artefact", artifact))
+    val scopeFilter: Bson             = scopes.fold[Bson](BsonDocument())(ss => or(ss.map(scope => equal(s"${scope.asString}Flag", value = true)): _*))
+    val repoFilter: Bson              = nin("repoType", Service.asString)
+
+    val filters = Seq(groupFilter, artifactFilter, Some(repoFilter), Some(scopeFilter)).flatten
+
+    collection
+      .find(if (filters.isEmpty) BsonDocument() else Filters.and(filters: _*))
+      .toFuture()
+  }
+
+  def findServicesByDeployment(
```

Comment:
Is this meant to replace `DerivedServiceDependenciesRepository` which is still here? The controller can probably just send them to the existing repository for now?